sale_rental/models/stock.py

```diff
@@ -0,0 +1,239 @@
+                "warehouse %s") % self.name)
+        rental_pull_rule = {
+            'name': self._format_rulename(
+                self.rental_in_location_id,
+                self.rental_out_location_id, ''),
+            'location_src_id': self.rental_in_location_id.id,
+            'location_id': self.rental_out_location_id.id,
+            'route_id': rental_route.id,
+            'action': 'pull',
+            'picking_type_id': self.out_type_id.id,
+            'warehouse_id': self.id,
+        }
+        rental_push_rule = {
+            'name': self._format_rulename(
+                self.rental_out_location_id,
+                self.rental_in_location_id, ''),
+            'location_src_id': self.rental_out_location_id.id,
+            'location_id': self.rental_in_location_id.id,
+            'route_id': rental_route.id,
+            'action': 'push',
+            'picking_type_id': self.in_type_id.id,
+            'warehouse_id': self.id,
+        }
+        customer_loc = self.env.ref('stock.stock_location_customers')
+        sell_rented_product_pull_rule = {
+            'name': self._format_rulename(
+                self.rental_out_location_id, customer_loc, ''),
+            'location_src_id': self.rental_out_location_id.id,
+            'location_id': customer_loc.id,
+            'route_id': sell_rented_product_route.id,
+            'action': 'pull',
+            'picking_type_id': self.out_type_id.id,
+            'warehouse_id': self.id,
+        }
+        res = [
+            rental_pull_rule,
+            rental_push_rule,
+            sell_rented_product_pull_rule,
+            ]
+        return res
+
+    def _create_rental_locations(self):
+        slo = self.env['stock.location']
+        for wh in self:
+            # create stock locations
+            if not wh.rental_view_location_id:
+                view_loc = slo.with_context(lang='en_US').search([
+                    ('name', 'ilike', 'Rental'),
+                    ('location_id', '=', wh.view_location_id.id),
+                    ('usage', '=', 'view')], limit=1)
+                if not view_loc:
+                    view_loc = slo.create({
+                        'name': _('Rental'),
+                        'location_id': wh.view_location_id.id,
+                        'usage': 'view',
+                    })
+                    logger.debug(
+                        'New view rental stock location created ID %d',
+                        view_loc.id)
+                wh.rental_view_location_id = view_loc.id
+            if not wh.rental_in_location_id:
+                in_loc = slo.with_context(lang='en_US').search([
+                    ('name', 'ilike', 'Rental In'),
+                    ('location_id', '=', wh.rental_view_location_id.id),
+                    ], limit=1)
+                if not in_loc:
+                    in_loc = slo.create({
+                        'name': _('Rental In'),
```

Comment:
Just a comment, we are searching for record with `with_context(lang='en_US')` but creating new one with lang

Reply:
fixed